README.md

```diff
@@ -153,34 +156,88 @@ Loading...
   ... the actual gallery...
 </template>
 ```
 
-#### Details of `<template for=... mode...>`
+#### Details of `<template for=...>`
+
+1. The `for` attribute behaves like `<label for>`, pointing to an ID.
+1. When a `<template for>` is discovered, its contents are streamed into its target position.
+1. The template itself is just a DOM directive. It doesn't stay in the DOM after being processed.
+1. Potentially, we can make this template point to a *position* in the DOM rather than an ID, e.g. one or two CSS selectors and replacing everything between them.
+   If we do that, the template does nothing and is discarded if it doesn't have anywhere to stream into.
 
-1. The `for` attribute behaves like `<label for>`, pointing to an ID
-1. `mode` can be `replace`, `append` or `prepend`, and be extended in the future to include things like `merge`, `loading` and `error`.
-1. In general, the `<template>` itself does not stay in the DOM, kind of like `<template shadowrootmode>`. However the details of this are TBD.
-1. When a `<template for>` is discovered, its contents are streamed into its target element.
- 
 This is, in essence, a proposed solution for https://github.com/whatwg/html/issues/2791. 
 
-### Part 2: Composable app fragments (views)
+### Part 2: URL matching with views & routes
+
+To effectively match between parts of the document and the state of the app, there is commonly a relationship between the document's URL and active/visible components of the app.
+This is commonly referred to in frameworks as "routes".
+
+The proposal here is to make routes a first-class citizen in HTML, and using that (at first) for two things:
+1. Wrapping certain parts of the documents that belong to a certain route with a `<view>` element that describes that relationship.
+2. CSS at-rule conditions that get activated by route-matching, similar to media-queries:
+
+```html
+<!-- Similar shape to importmap -->
+<script type=routemap>
+  {
+    "movies": { "pattern": { "pathname": "/movies/:movie_id" } },
+    "about":  { "pattern": "/about" },
+    "home":   { "pattern": ["/home", "/"] }
+  ]
+</script>
+
+<style>
+@route home {
+  nav {
+    display: none;
+  }
+}
+</style>
+
+<body>
+   <view match=home>
+   </view>
+   <view match=about>
+   </view>
+   <view match=movies>
+   </view>
+</body>
+```
+
+#### Details of `<view match=...>` and `<script type=routemap>`
+
+1. Every route can be one or more `URLPattern` (more precisely, the parameters to the `URLPattern` constructor)
+1. A `view` can be described by using CSS. It is merely an element that has `content-visibility: hidden` when the route doesn't match.
+   The choice to make it an element makes it clear semantically that this part of the document is an app fragment.
+1. Views can be extended in the future to support a per-view URL ("HTML includes"), and have additional CSS selectors for links that target a particular route.
+1. Multiple views that match the same route can be present at the same time
```

Comment:
Similarly, it's possible for multiple routes to match at the same time unless carefully constructed to avoid this. Or should the first or most specific route win so that only one route is considered matching at once?

Reply:
Yes! Added a note about this. Though we can discuss the specifics.